### PR TITLE
[AIRFLOW-4790] Change dag processing stats to timers

### PR DESCRIPTION
**Pull request by andrewhharmon**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following issues and references them in the PR title. 
  - https://issues.apache.org/jira/browse/AIRFLOW-4790

### Description

- [x] the stasd metrics for dag processing should be changed from gauges to timers

### Tests

- [x] scheduler job already has unit test

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - updated metrics docs

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Review by feng-tao — DISMISSED** [review]
I think you are right. It should be timers.

File: airflow/utils/dag_processing.py [hunk 1]
```diff
@@ -984,7 +984,7 @@ def _log_file_processing_stats(self, known_file_paths):
             file_name = os.path.basename(file_path)
             file_name = os.path.splitext(file_name)[0].replace(os.sep, '.')
             if last_runtime:
-                Stats.gauge(
+                Stats.timing(
```
Comment (feng-tao) [inline airflow/utils/dag_processing.py:987]: actually, the stat calculates the time for the dag processing time in the latest/last iteration. If the data changes, the stat will change. So I think gauge works in this case. Could you share why you think gauge doesn't work?

Comment (andrewhharmon) [reply airflow/utils/dag_processing.py:987]: ah, i guess we are interested in tracking how long it takes to process a dag over time. so we can see if dag processing is slowing down over time

Comment (andrewhharmon) [reply airflow/utils/dag_processing.py:987]: maybe i don't fully understand how the scheduler works and what those metrics were counting. i think i understand why it's a gauge here. Maybe i need to add the metric we're interested in in another place. any thoughts? we want to see a line chart that shows if dag processing is taking a while to help alert if the scheduler is overwhelmed.

Comment (feng-tao) [reply airflow/utils/dag_processing.py:987]: +1, I think both stats are valid.

Comment (feng-tao) [reply airflow/utils/dag_processing.py:987]: @andrewhharmon actually I have a open pr which will instrument the dag loading/processing time. Is that what you want?

Comment (feng-tao) [reply airflow/utils/dag_processing.py:987]: https://github.com/apache/airflow/pull/5350

Comment (andrewhharmon) [reply airflow/utils/dag_processing.py:987]: yeah, that looks perfect. we can delete my PR. Thanks!

**Pull request closed without merging**